content/posts/0000-buildpack-philosophy-part-1.md

```diff
@@ -0,0 +1,256 @@
+consider how it might be broken down into parts that can be composed together
+to create more complex behaviors. In turn, these parts can be reused to enable
+other workflows that weren't originally foreseen.
+
+If you dig into many of our language family buildpacks ([.Net
+Core](https://github.com/paketo-buildpacks/dotnet-core),
+[Go](https://github.com/paketo-buildpacks/go),
+[Java](https://github.com/paketo-buildpacks/java), etc.) you'll notice that
+they are made up of many smaller buildpacks. Just [taking a
+look](https://github.com/paketo-buildpacks/nodejs/blob/33aad92fd908665122d9ca6345fcfd97fa3d86bb/buildpack.toml)
+at the NodeJS buildpack, we can see that it consists of 11 smaller buildpacks:
+
+* [Node Engine](https://github.com/paketo-buildpacks/node-engine)
+* [Node Start](https://github.com/paketo-buildpacks/node-start)
+* [NPM Install](https://github.com/paketo-buildpacks/npm-install)
+* [NPM Start](https://github.com/paketo-buildpacks/npm-start)
+* [Yarn](https://github.com/paketo-buildpacks/yarn)
+* [Yarn Install](https://github.com/paketo-buildpacks/yarn-install)
+* [Yarn Start](https://github.com/paketo-buildpacks/yarn-start)
+* [Procfile](https://github.com/paketo-buildpacks/procfile)
+* [Environment Variables](https://github.com/paketo-buildpacks/environment-variables)
+* [Image Labels](https://github.com/paketo-buildpacks/image-labels)
+* [CA Certificates](https://github.com/paketo-buildpacks/ca-certificates)
+
+Each one of these buildpacks performs a small role in the overall NodeJS
+buildpack. They provide dependencies, execute build tool processes, or
+configure the application to run in a containerized environment. They do their
+single part of the job well, and in a focused manner. For example, given a
+[Yarn](https://yarnpkg.com/)-based application, the Node Engine buildpack
+installs the NodeJS runtime, the Yarn buildpack installs the Yarn tool, the
+Yarn Install buildpack executes the `yarn install` build process, and the Yarn
+Start buildpack chooses the right command to start the application in the
+container.
+
+These buildpacks can be used over again in different combinations and in
+entirely different buildpack language ecosystems. Its pretty common for
+[Rails](http://rubyonrails.org/) apps to require a JavaScript runtime to
+compile their assets. The Ruby buildpack includes the Node Engine buildpack to
+provide that runtime. It works well because the Node Engine buildpack limits
+itself to performing only that one task.
+
+#### Be transparent and understandable
+
+Provide the buildpack user with transparent and understandable feedback during
+all phases of the buildpack lifecycle. When a buildpack fails to detect,
+specify why. When a buildpack fails to build, include as much logging
+information as is reasonable to help the user orient themselves in the build
+context at the point of failure. Prefer concise prose messages with clear
+phrasing over symbols or colors.
```

Comment:
I realize that you have phrased this with the word "prefer", but since our Java buildpacks use colored logging output perhaps we should leave this bit about using color out?